lifecycle/lifecycle-livedata-core-ktx-lint/src/main/java/androidx/lifecycle/lint/NonNullableMutableLiveDataDetector.kt

```diff
@@ -94,15 +96,26 @@ class NonNullableMutableLiveDataDetector : Detector(), UastScanner {
             }
 
             private fun getFieldTypeReference(element: KotlinUField): KtTypeReference? {
+                // If field has type reference, we need to use type reference
+                // Given the field `val liveDataField: MutableLiveData<Boolean> = MutableLiveData()`
+                // expression: `MutableLiveData<Boolean>()`
+                // argument: `Boolean`
+                val typeReference = element.sourcePsi
+                    ?.children
+                    ?.firstOrNull { it is KtTypeReference } as? KtTypeReference
+                val typeArgument = typeReference?.typeElement?.typeArgumentsAsTypes?.singleOrNull()
+                if (typeArgument != null) {
+                    return typeArgument
+                }
+
                 // We need to extract type from the expression
                 // Given the field `val liveDataField = MutableLiveData<Boolean>()`
                 // expression: `MutableLiveData<Boolean>()`
                 // argument: `Boolean`
-                val expression = element.sourcePsi
+                val callExpression = element.sourcePsi
```

Comment:
keep this val name as `expression` or change the comment above. Either works.

Reply:
Reverted name change.